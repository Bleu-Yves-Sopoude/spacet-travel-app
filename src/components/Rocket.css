.rocketsList {
  margin: 20px;
}

.rocketsList li {
  display: flex;
  gap: 40px;
  height: 100%;
  margin: 50px 0;
}

.rocketsList li .rocketImgageContainer {
  max-width: 250px;
  width: 100%;
  overflow: hidden;
  border: 1px solid rgba(0, 0, 0, 0.111);
}

.rocketsList li .rocketImgageContainer img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.rocketsList li .rocketDescription p {
  max-width: 100%;
  width: 100%;
  line-height: 1.7rem;
  color: #334155;
  margin-top: 20px;
}

.rocketsList li .rocketDescription p span {
  background-color: #0af;;
  padding: 4px 10px;
  margin-right: 3px;
  color: white;
  border-radius: 5px;
}

.rocketsList li .rocketDescription h2 {
  color: #334155;
}

.rocketsList li .rocketDescription button {
  margin-top: 20px;
  padding: 14px 40px;
  border-radius: 5px;
  background: #4f46e5;
  border: none;
  color: white;
  cursor: pointer;
}

.rocketsList li .rocketDescription button.cancel {
  background-color: rgba(128, 128, 128, 0.21);
  color: gray;
  border: 1px solid rgba(128, 128, 128, 0.514);
}
