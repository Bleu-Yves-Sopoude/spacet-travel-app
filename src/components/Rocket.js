import { React, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rocketsSlice';

const Rocket = () => {
  const rocketsdata = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  return (
    <div>
      {rocketsdata.loading ? (
        <div>Loading ...</div>
      ) : (
        <ul className="rocketsList">
          {rocketsdata.data.map((e) => (
            <>
              <li key={e.id}>
                {e.flickr_images && <img src={e.flickr_images} alt={e.name} />}
                <h2>{e.name}</h2>
                <p>
                  {!e.reserved && <p style={{background:"red", padding:"40px"}}>Reserved</p>} {e.description}
                </p>
              </li>
              <hr/>
            </>

          ))}
        </ul>
      )}
    </div>
  );
};

export default Rocket;
