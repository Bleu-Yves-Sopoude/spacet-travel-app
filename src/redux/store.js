import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionsSlice';
import rocketsReducer from './rockets/rocketsSlice.js'

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets:rocketsReducer,
  },
});

export default store;
