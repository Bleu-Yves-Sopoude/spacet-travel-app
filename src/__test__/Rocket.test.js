import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import Rocket from '../components/Rocket';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);
describe('MyRocketss', () => {
  it('render Rockets Components', () => {
    render(<Rocket />);
    expect(Rocket).toMatchSnapshot();
  });
});
