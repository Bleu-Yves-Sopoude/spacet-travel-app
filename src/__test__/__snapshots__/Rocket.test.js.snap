// Jest Snapshot v1, https://goo.gl/fbAQLP



exports[`MyRocketss render Rockets Components 1`] = `[Function]`;
